apps/argus/src/chain/ethereum.rs

```diff
@@ -0,0 +1,317 @@
+use {
+    crate::{
+        api::ChainId,
+        chain::reader::{self, BlockNumber, PulseReader, RequestedWithCallbackEvent},
+        config::EthereumConfig,
+    },
+    anyhow::{anyhow, Context, Result},
+    axum::async_trait,
+    ethers::{
+        abi::RawLog,
+        contract::{abigen, EthLogDecode},
+        core::types::Address,
+        middleware::{gas_oracle::GasOracleMiddleware, SignerMiddleware},
+        prelude::JsonRpcClient,
+        providers::{Http, Middleware, Provider},
+        signers::{LocalWallet, Signer},
+        types::U256,
+    },
+    fortuna::eth_utils::{
+        eth_gas_oracle::EthProviderOracle,
+        legacy_tx_middleware::LegacyTxMiddleware,
+        nonce_manager::NonceManagerMiddleware,
+        traced_client::{RpcMetrics, TracedClient},
+    },
+    std::sync::Arc,
+};
+
+// TODO: Programmatically generate this so we don't have to keep committed ABI in sync with the
+// contract in the same repo.
+abigen!(
+    Pulse,
+    "../../target_chains/ethereum/contracts/out/IPulse.sol/IPulse.abi.json"
```

Comment:
This seems like an external dependency. either manually move it in, or add it to the build script of this repo; also mention it in README

Reply:
and i'm wondering what built command you used, i needed to run `forge build --extra-output-files abi`